src/Extensions/TimedScopes/Internal/ActivityObserversIntializer.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.TimedScopes
+{
+	internal sealed class ActivityObserversIntializer : IHostedService, IObserver<DiagnosticListener>, IObserver<KeyValuePair<string, object>>
+	{
+		private readonly IActivityStartObserver[] m_activityStartObservers;
+		private readonly IActivityStopObserver[] m_activityStopObservers;
+		private readonly LinkedList<IDisposable> m_disposables;
+		private IDisposable? m_observerLifetime;
+
+		public ActivityObserversIntializer(
+			IEnumerable<IActivityStartObserver> activityStartObservers,
+			IEnumerable<IActivityStopObserver> activityStopObservers)
+		{
+			m_activityStartObservers = activityStartObservers.ToArray();
+			m_activityStopObservers = activityStopObservers.ToArray();
+			m_disposables = new LinkedList<IDisposable>();
+		}
+
+		public Task StartAsync(CancellationToken cancellationToken)
+		{
+			m_observerLifetime = DiagnosticListener.AllListeners.Subscribe(this);
+			return Task.CompletedTask;
+		}
+		
+		public Task StopAsync(CancellationToken cancellationToken)
+		{
+			foreach (IDisposable disposable in m_disposables)
+			{
+				disposable.Dispose();
+			}
```

Comment:
>	} [](start = 2, length = 2)

Nitpick: It's often considered best practice to skip a blank line after `}`

Reply:
Added, thank you